selfupdate/detect.go

```diff
@@ -115,13 +131,13 @@ func findReleaseAndAsset(rels []*github.RepositoryRelease, targetVersion string)
 // where 'foo' is a command name. '-' can also be used as a separator. File can be compressed with zip, gzip, zxip, tar&zip or tar&zxip.
 // So the asset can have a file extension for the corresponding compression format such as '.zip'.
 // On Windows, '.exe' also can be contained such as 'foo_windows_amd64.exe.zip'.
-func (up *Updater) DetectLatest(slug string) (release *Release, found bool, err error) {
-	return up.DetectVersion(slug, "")
+func (up *Updater) DetectLatest(slug string, opts ...Option) (release *Release, found bool, err error) {
+	return up.DetectVersion(slug, "", opts...)
```

Comment:
This library provides customization points as fields of `Config` struct rather than adding arguments. For example. you can set a validator. I adopted this design to keep methods of `Updater` struct simple as much as possible.

https://github.com/rhysd/go-github-selfupdate/blob/master/selfupdate/updater.go#L22

Would you prefer adding a field to `Config` struct rather than adding argument? Please keep compiled `Regexp` instance in `Updater` struct.

```go
type Config struct {
	// APIToken represents GitHub API token. If it's not empty, it will be used for authentication of GitHub API
	APIToken string
	// EnterpriseBaseURL is a base URL of GitHub API. If you want to use this library with GitHub Enterprise,
	// please set "https://{your-organization-address}/api/v3/" to this field.
	EnterpriseBaseURL string
	// EnterpriseUploadURL is a URL to upload stuffs to GitHub Enterprise instance. This is often the same as an API base URL.
	// So if this field is not set and EnterpriseBaseURL is set, EnterpriseBaseURL is also set to this field.
	EnterpriseUploadURL string
	// Validator represents types which enable additional validation of downloaded release.
	Validator Validator
	// ...
	TagFilters []string
}

type Updater struct {
	api    *github.Client
	apiCtx context.Context
	validator Validator
	filters []regexp.Regexp 
}
```

Reply:
ok your place, your rules. I don't mind. I was really in a hurry to implement and unlock my own job. Not a problem to make it this way.